File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        
        if(nums.size()==1) return nums[0];

        sort(nums.begin(),nums.end());
        int count=1,threshold=nums.size()/2;

        for(int i=1;i<nums.size();i++){
            
            if(nums[i-1]==nums[i]) count++;
            
            if(count>threshold) return nums[i];
            
            else if(nums[i]!=nums[i-1]) count=1;
        }
        return -1;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int element=-1,count=0;

        for(int num:nums){
            if(count==0) element=num;
           if(element==num) count++;
           else count--;
        }

        int totalFreq=0;
        for(int num:nums){
            if(num==element) totalFreq++;
        }

        return totalFreq>nums.length/2 ? element : -1;
    }
}